build(aio): add initial support for generating guides

petebacondarwin opened the pull request:
This PR demonstrates how we are going to render the guide docs.
It takes a single cookbook article (and its associated examples) and generates HTML from it.

This uses the Rho markdown rendering engine, which has a few bugs. But I have two open PRs to fix those right now: https://github.com/inca/rho/pulls

petebacondarwin committed:
docs(aio): add examples to test cookbook doc

petebacondarwin committed:
docs(aio): add cookbook doc

petebacondarwin commented on tools/docs/angular.io-package/index.js:
```diff
@@ -47,13 +48,15 @@ module.exports =
         .config(function(checkAnchorLinksProcessor, log) {
           // TODO: re-enable
           checkAnchorLinksProcessor.$enabled = false;
+          // log.level = 'silly';
```
oops need to remove this.

petebacondarwin commented on tools/docs/angular.io-package/index.js:
```diff
@@ -47,13 +48,15 @@ module.exports =
             readTypeScriptModules, readFilesProcessor, collectExamples, generateKeywordsProcessor) {
 
           // API files are typescript
+          readTypeScriptModules.$enabled = false;
```
oops need to remove this

petebacondarwin committed:
build(aio): support guide authoring

This commit implements various tags, inline tags and changes the markdown
renderer to use Rho. This enables us to generate guide type documents.

petebacondarwin committed:
docs(zone): fix whitespace around backtick code

If there is no leading empty line then the markdown renderers get confused.

wardbell commented:
We talked about this but thought I should leave a note here. Many of these files are sample boilerplate that should not be checked in here.  Pete is going to purify this PR and bring the sample generation-and-maintenance infrastructure over from angular.io in a prudent way.

I'm not sure if this PR yet reflects these intentions as the folder structure may be different here in angular/angular

IgorMinar commented:
@petebacondarwin CI is failing.

@wardbell I'm fine merging this if necessary for iterating. all this content stuff will be replaced anyway once we merge the repos.

IgorMinar reviewed (approved):
please fix the CI failure and respond to @wardbell comment

petebacondarwin commented:
@IgorMinar I have addressed the CI problem.
@wardbell I have removed the boilerplate files. It may be that we move the examples to another folder later once your work on the information architecture is complete.
